### PR TITLE
OTA-4241: fix aktualizr brewing formula

### PR DESCRIPTION
**Pull request by mike-sul**
Signed-off-by: Mykhaylo Sul <myk.sul@gmail.com>

### CODE REVIEW

File: aktualizr.rb [hunk 1]
```diff
@@ -1,32 +1,38 @@
...
     cellar :any
-    sha256 "021255fc3b2b7409ab649adeb0abdb71efec564519ca7d0a6ad36d92abc9f8b6" => :mojave
+    sha256 "021255fc3b2b7409ab649adeb0abdb71efec564519ca7d0a6ad36d92abc9f8b6" => :catalina
   end
 
   def install
     mkdir "build" do
-      system "cmake", "..", *std_cmake_args
-      system "make"
+      system "cmake", "..", "-DBoost_USE_MULTITHREADED=ON", "-DOPENSSL_SSL_LIBRARY=/usr/local/opt/openssl@1.1/lib/libssl.dylib", "-DOPENSSL_CRYPTO_LIBRARY=/usr/local/opt/openssl@1.1/lib/libcrypto.dylib", *std_cmake_args
```
Comment (tkfu) [inline aktualizr.rb]:
```suggestion
      system "cmake", "..", "-DBoost_USE_MULTITHREADED=ON", "-DOPENSSL_SSL_LIBRARY=#{Formula["openssl@1.1"].opt_prefix}/lib/libssl.dylib", "-DOPENSSL_CRYPTO_LIBRARY=#{Formula["openssl@1.1"].opt_prefix}/lib/libcrypto.dylib", *std_cmake_args
```

Just to make sure it works for users with weird directory layouts.

File: aktualizr-dev.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  end
+
+  def install
+    mkdir "build" do
+      system "cmake", "..", "-DBoost_USE_MULTITHREADED=ON", "-DOPENSSL_SSL_LIBRARY=/usr/local/opt/openssl@1.1/lib/libssl.dylib", "-DOPENSSL_CRYPTO_LIBRARY=/usr/local/opt/openssl@1.1/lib/libcrypto.dylib", *std_cmake_args
+
+      system "make",  "-j#{ENV.make_jobs}"
+
+      system "mkdir", "-p", "#{bin}"
+      system "cp", "./src/aktualizr_primary/aktualizr", "#{bin}/aktualizr"
```
Comment (tkfu) [inline aktualizr-dev.rb]:
```suggestion
      system "mkdir", "-p", "#{lib}"

      system "cp", "./src/aktualizr_primary/aktualizr", "#{bin}/aktualizr_dev"
      system "cp", "./src/libaktualizr/libaktualizr_lib.dylib", "#{lib}"
```

Better to give the binary a different name, so you can run both a stable one and the bleeding-edge one on your system if desired.

File: aktualizr-dev.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  depends_on "python3" => :build
+
+  bottle do
+    cellar :any
+    sha256 "021255fc3b2b7409ab649adeb0abdb71efec564519ca7d0a6ad36d92abc9f8b6" => :catalina
+  end
+
+  def install
+    mkdir "build" do
+      system "cmake", "..", "-DBoost_USE_MULTITHREADED=ON", "-DOPENSSL_SSL_LIBRARY=/usr/local/opt/openssl@1.1/lib/libssl.dylib", "-DOPENSSL_CRYPTO_LIBRARY=/usr/local/opt/openssl@1.1/lib/libcrypto.dylib", *std_cmake_args
```
Comment (tkfu) [inline aktualizr-dev.rb]:
```suggestion
      system "cmake", "..", "-DBoost_USE_MULTITHREADED=ON", "-DOPENSSL_SSL_LIBRARY=#{Formula["openssl@1.1"].opt_prefix}/lib/libssl.dylib", "-DOPENSSL_CRYPTO_LIBRARY=#{Formula["openssl@1.1"].opt_prefix}/lib/libcrypto.dylib", *std_cmake_args
```
Same minor fix as the other formula.

File: aktualizr-dev.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
+class AktualizrDev < Formula
+  desc "C++ Client for HERE OTA Connect"
+  homepage ""
+  version = "2020.2"
+  url "https://github.com/advancedtelematic/aktualizr.git", :using => :git, :tag => "#{version}"
+  sha256 "782fa343c85be455d6e51bd774f3244e0dad093989ac9bb1d96215785f7e7314"
```
Comment (tkfu) [inline aktualizr-dev.rb:6]: For this one, wouldn't it make more sense to build master?

Comment (tkfu) [reply aktualizr-dev.rb:6]: Actually, come to think of it, homebrew has a `--HEAD` option. We could just add `head "https://github.com/advancedtelematic/aktualizr.git"` to the main formula, and then do `brew install --fetch-HEAD --HEAD aktualizr` if/when we want to test the master build.

**Review by tkfu — CHANGES_REQUESTED** [review]
Nice! Thanks for fixing the formula, and the dynamic libraries issue.

I think we can have an aktualizr-dev formula if we think it's useful to have two versions installed in parallel, or we can just add the head option to the main formula if all we really care about is testing.

File: aktualizr-dev.rb (hunk 4 above)
Comment (mike-sul) [reply aktualizr-dev.rb:6]: My initial intention was to build a head of branch and this `head` feature didn't work for branch for me.

Comment (tkfu) [reply aktualizr-dev.rb:6]: Okay. I still don't understand what this formula is for; after all, it's building from a tagged release just like the main aktualizr recipe. But if you find it useful, I don't see any reason not to have it here.

**Review by tkfu — APPROVED** [review]

**Pull request merged**